projects/gloo/pkg/plugins/headers/plugin.go

```
@@ -33,18 +38,22 @@ var (
 )
 
 // Puts Header Manipulation config on Routes, VirtualHosts, and Weighted Clusters
-type plugin struct{}
+type plugin struct {
+	enforceMatchingNamespaces bool
+}
 
 func NewPlugin() *plugin {
-	return &plugin{}
+	enforceMatch, _ := strconv.ParseBool(os.Getenv(api_conversion.MatchingNamespaceEnv))
```

instead of swallowing the error here, would it be better to check it during translation and error if there's an invalid value? just thinking of the case where user might have made a typo in their helm values and not realize it

that said since this already merged into main, i'll approve so there's no drift

---

main hasnt released yet so pehaps we fix it here and then throw up a quick change against main?